### PR TITLE
feat(cfc library): add cfc library

### PR DESCRIPTION
**Pull request by VERZUOL1**

### CODE REVIEW

File: astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx [hunk 1]
```diff
@@ -88,18 +93,22 @@ const schema = yup.object().shape({
...
   onDelete,
   onSaveToDaos,
   disabled,
   editable,
+  name,
+  isEnabled,
 }) => {
-  const { dao } = daoContext;
-  const { config } = template;
+  // const { dao } = daoContext;
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx]: Can we drop comments?

File: astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx [hunk 2]
```diff
@@ -39,15 +38,21 @@ ace.config.set(
...
 interface Props {
-  daoContext: DaoContext;
-  template: ProposalTemplate;
-  accountDaos: DaoFeedItem[];
-  onUpdate: (id: string, data: TemplateUpdatePayload) => void;
-  onDelete: (id: string) => void;
+  daoId?: string;
+  templateId?: string;
+  flagCover?: string;
+  // daoContext: DaoContext;
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx]: Can we drop comments?

File: astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/CustomFunctionCallTemplatesPageContent.tsx [hunk 3]
```diff
@@ -105,15 +105,21 @@ export const CustomFunctionCallTemplatesPageContent: FC<Props> = ({
               .map(item => (
                 <CustomFcTemplateCard
                   key={`${item.id}_${item.updatedAt}`}
-                  daoContext={daoContext}
+                  // daoContext={daoContext}
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/CustomFunctionCallTemplatesPageContent.tsx]: Can we drop comments?

File: astro_2.0/features/pages/cfcLibrary/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+import { SputnikHttpService } from 'services/sputnik';
+import { LIST_LIMIT_DEFAULT } from 'services/sputnik/constants';
+import { PaginationResponse } from 'types/api';
+import { SharedProposalTemplate } from 'types/proposalTemplate';
+import { useRouter } from 'next/router';
+import { useDebounceEffect } from 'hooks/useDebounceUpdateEffect';
+
+export type PageData = PaginationResponse<SharedProposalTemplate[]>;
+
+export function useCfcLibraryData(): {
```
Comment (AlexeyPonomarenko) [inline astro_2.0/features/pages/cfcLibrary/hooks.ts:12]: Can we split that into 2 separate files - `useCfcLibraryData` and `useSharedTemplatePageData` and place them into `hooks` folder?

**Review by AlexeyPonomarenko — DISMISSED** [review]

File: astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx (hunk 1 above)
Comment (VERZUOL1) [reply astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/components/CustomFcTemplateCard/CustomFcTemplateCard.tsx]: Fixed

File: astro_2.0/features/pages/cfcLibrary/hooks.ts (hunk 4 above)
Comment (VERZUOL1) [reply astro_2.0/features/pages/cfcLibrary/hooks.ts:12]: I would like to keep these 2 hook at the same place. They are now application wide used, they are very special to specific page and defined to keep page logic only.

**Review by dmytro-almighty — DISMISSED** [review]

File: astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/CustomFunctionCallTemplatesPageContent.tsx (hunk 3 above)
Comment (VERZUOL1) [reply astro_2.0/features/pages/nestedDaoPagesContent/CustomFunctionCallTemplatesPageContent/CustomFunctionCallTemplatesPageContent.tsx]: done

**Review by dmytro-almighty — APPROVED** [review]

**Pull request merged**